### ISSUE TITLE
Questions about OCRBench v1

### ISSUE THREAD

**Issue by lyccyl1** [Yuliang-Liu/MultimodalOCR#65]
Hi, I'm trying to eval Qwen2.5-VL-7B-Instruct in OCRBench. But I get the final score only 12. Does this have any issues? 
Here is my pip list: 
`absl-py                       2.3.1
accelerate                    1.8.1
certifi                       2025.6.15
charset-normalizer            3.4.2
contourpy                     1.3.2
cycler                        0.12.1
einops                        0.8.1
filelock                      3.18.0
fonttools                     4.58.5
fsspec                        2025.5.1
grpcio                        1.73.1
hf-xet                        1.1.5
huggingface-hub               0.33.2
idna                          3.10
Jinja2                        3.1.6
kiwisolver                    1.4.8
Markdown                      3.8.2
MarkupSafe                    3.0.2
matplotlib                    3.10.3
mpmath                        1.3.0
networkx                      3.4.2
numpy                         2.2.6
nvidia-cublas-cu11            11.11.3.6
nvidia-cublas-cu12            12.4.5.8
nvidia-cuda-cupti-cu11        11.8.87
nvidia-cuda-cupti-cu12        12.4.127
nvidia-cuda-nvrtc-cu11        11.8.89
nvidia-cuda-nvrtc-cu12        12.4.127
nvidia-cuda-runtime-cu11      11.8.89
nvidia-cuda-runtime-cu12      12.4.127
nvidia-cudnn-cu11             9.1.0.70
nvidia-cudnn-cu12             9.1.0.70
nvidia-cufft-cu11             10.9.0.58
nvidia-cufft-cu12             11.2.1.3
nvidia-curand-cu11            10.3.0.86
nvidia-curand-cu12            10.3.5.147
nvidia-cusolver-cu11          11.4.1.48
nvidia-cusolver-cu12          11.6.1.9
nvidia-cusparse-cu11          11.7.5.86
nvidia-cusparse-cu12          12.3.1.170
nvidia-cusparselt-cu12        0.6.2
nvidia-nccl-cu11              2.20.5
nvidia-nccl-cu12              2.21.5
nvidia-nvjitlink-cu12         12.4.127
nvidia-nvtx-cu11              11.8.86
nvidia-nvtx-cu12              12.4.127
packaging                     25.0
pillow                        11.3.0
pip                           25.1
protobuf                      6.31.1
psutil                        7.0.0
pyparsing                     3.2.3
python-dateutil               2.9.0.post0
PyYAML                        6.0.2
regex                         2024.11.6
requests                      2.32.4
safetensors                   0.5.3
scipy                         1.15.3
setuptools                    78.1.1
six                           1.17.0
sympy                         1.13.1
tensorboard                   2.19.0
tensorboard-data-server       0.7.2
tiktoken                      0.9.0
tokenizers                    0.21.2
torch                         2.4.0+cu118
torchaudio                    2.4.0+cu118
torchvision                   0.19.0+cu118
tqdm                          4.67.1
transformers                  4.51.3
transformers-stream-generator 0.0.4
triton                        3.0.0        1
typing_extensions             4.14.1
urllib3                       2.5.0
Werkzeug                      3.1.3
wheel                         0.45.1`

I'm trying the code below, which is modified from /MultimodalOCR/OCRBench/scripts/qwenvl.py.

`import json
from argparse import ArgumentParser
import torch
import os
import json
from tqdm import tqdm
from PIL import Image
import math
import multiprocessing
from multiprocessing import Pool, Queue, Manager
from transformers import AutoModelForCausalLM, AutoTokenizer
from transformers import Qwen2VLForConditionalGeneration
def split_list(lst, n):
    length = len(lst)
    avg = length // n  # 每份的大小
    result = []  # 存储分割后的子列表
    for i in range(n - 1):
        result.append(lst[i*avg:(i+1)*avg])
    result.append(lst[(n-1)*avg:])
    return result
def save_json(json_list,save_path):
    with open(save_path, 'w') as file:
        json.dump(json_list, file,indent=4)
def _get_args():
    parser = ArgumentParser()
    parser.add_argument("--image_folder", type=str, default="/MultimodalOCR/OCRBench/images/data")
    parser.add_argument("--output_folder", type=str, default="./results")
    parser.add_argument("--OCRBench_file", type=str, default="/MultimodalOCR/OCRBench/OCRBench/OCRBench.json")
    parser.add_argument("--model_path", type=str, default="/data0/user/ycliu/models/qwen2.5-vl-7b")
    parser.add_argument("--save_name", type=str, default="/MultimodalOCR/OCRBench/qwenvl")
    parser.add_argument("--num_workers", type=int, default=1)
    args = parser.parse_args()
    return args
OCRBench_score = {"Regular Text Recognition":0,"Irregular Text Recognition":0,"Artistic Text Recognition":0,"Handwriting Recognition":0,
"Digit String Recognition":0,"Non-Semantic Text Recognition":0,"Scene Text-centric VQA":0,"Doc-oriented VQA":0,
"Key Information Extraction":0,"Handwritten Mathematical Expression Recognition":0}
AllDataset_score = {"IIIT5K":0,"svt":0,"IC13_857":0,"IC15_1811":0,"svtp":0,"ct80":0,"cocotext":0,"ctw":0,"totaltext":0,"HOST":0,"WOST":0,"WordArt":0,"IAM":0,"ReCTS":0,"ORAND":0,"NonSemanticText":0,"SemanticText":0,
"STVQA":0,"textVQA":0,"ocrVQA":0,"ESTVQA":0,"ESTVQA_cn":0,"docVQA":0,"infographicVQA":0,"ChartQA":0,"ChartQA_Human":0,"FUNSD":0,"SROIE":0,"POIE":0,"HME100k":0}
num_all = {"IIIT5K":0,"svt":0,"IC13_857":0,"IC15_1811":0,"svtp":0,"ct80":0,"cocotext":0,"ctw":0,"totaltext":0,"HOST":0,"WOST":0,"WordArt":0,"IAM":0,"ReCTS":0,"ORAND":0,"NonSemanticText":0,"SemanticText":0,
"STVQA":0,"textVQA":0,"ocrVQA":0,"ESTVQA":0,"ESTVQA_cn":0,"docVQA":0,"infographicVQA":0,"ChartQA":0,"ChartQA_Human":0,"FUNSD":0,"SROIE":0,"POIE":0,"HME100k":0}

def eval_worker(args, data, eval_id, output_queue):
    print(f"Process {eval_id} start.")
    checkpoint = args.model_path
    # register before xxxx.from_pretrained
    from transformers import AutoModelForCausalLM, Qwen2VLConfig, Qwen2VLForConditionalGeneration,AutoModelForVision2Seq
    AutoModelForCausalLM.register(config_class=Qwen2VLConfig, model_class=Qwen2VLForConditionalGeneration)
    model = AutoModelForVision2Seq.from_pretrained(
        checkpoint, device_map=f'cuda:{eval_id}', trust_remote_code=True).eval()
    tokenizer = AutoTokenizer.from_pretrained(checkpoint,
                                              trust_remote_code=True)
    tokenizer.padding_side = 'left'
    # tokenizer.pad_token_id = tokenizer.eod_id
    for i in tqdm(range(len(data))):
        img_path = os.path.join(args.image_folder, data[i]['image_path'])
        qs = data[i]['question']
        # query = f'<img>{img_path}</img> {qs} Answer: '
        query = f'<img>{img_path}</img>{qs} Answer:'
        input_ids = tokenizer(query, return_tensors='pt', padding='longest')
        attention_mask = input_ids.attention_mask
        input_ids = input_ids.input_ids
        
        pred = model.generate(
        input_ids=input_ids.to(f'cuda:{eval_id}'),
        attention_mask=attention_mask.to(f'cuda:{eval_id}'),
        do_sample=False,
        num_beams=1,
        max_new_tokens=100,
        min_new_tokens=1,
        length_penalty=1,
        num_return_sequences=1,
        output_hidden_states=True,
        use_cache=True,
        # pad_token_id=tokenizer.eod_id,
        # eos_token_id=tokenizer.eod_id,
        )
        response = tokenizer.decode(pred[0][input_ids.size(1):].cpu(), skip_special_tokens=True).strip()
        data[i]['predict'] = response
    output_queue.put({eval_id: data})
    print(f"Process {eval_id} has completed.")

if __name__=="__main__":
    multiprocessing.set_start_method('spawn')
    args = _get_args()
    if os.path.exists(os.path.join(args.output_folder,f"{args.save_name}.json")):
        data_path = os.path.join(args.output_folder,f"{args.save_name}.json")
        print(f"output_path:{data_path} exist! Only generate the results that were not generated in {data_path}.")
    else:
        data_path = args.OCRBench_file

    with open(data_path, "r") as f:
        data = json.load(f)
        # print(data)
    
    data_list = split_list(data, args.num_workers)
    # print(data_list)
    output_queue = Manager().Queue()

    # 多线程
    # pool = Pool(processes=args.num_workers)
    # for i in range(len(data_list)):
    #     pool.apply_async(eval_worker, args=(args, data_list[i], i, output_queue))
    # pool.close()
    # pool.join()
    
    # 单线程
    eval_worker(args, data_list[0],0, output_queue)

    results = {}
    while not output_queue.empty():
        result = output_queue.get()
        results.update(result)
    print(results)
    data = []
    for i in range(len(data_list)):
        data.extend(results[i])

    for i in range(len(data)):
        data_type = data[i]["type"]
        dataset_name = data[i]["dataset_name"]
        answers = data[i]["answers"]
        if data[i].get('predict',0)==0:
            continue
        predict = data[i]['predict']
        data[i]['result'] = 0
        if dataset_name == "HME100k":
            if type(answers)==list:
                for j in range(len(answers)):
                    answer = answers[j].strip().replace("\n"," ").replace(" ","")
                    predict = predict.strip().replace("\n"," ").replace(" ","")
                    if answer in predict:
                        data[i]['result'] = 1
            else:
                answers = answers.strip().replace("\n"," ").replace(" ","")
                predict = predict.strip().replace("\n"," ").replace(" ","")
                if answers in predict:
                    data[i]['result'] = 1
        else:
            if type(answers)==list:
                for j in range(len(answers)):
                    answer = answers[j].lower().strip().replace("\n"," ")
                    predict = predict.lower().strip().replace("\n"," ")
                    if answer in predict:
                        data[i]['result'] = 1
            else:
                answers = answers.lower().strip().replace("\n"," ")
                predict = predict.lower().strip().replace("\n"," ")
                if answers in predict:
                    data[i]['result'] = 1
    save_json(data, os.path.join(args.output_folder,f"{args.save_name}.json"))
    if len(data)==1000:
        for i in range(len(data)):
            if data[i].get("result",100)==100:
                continue
            OCRBench_score[data[i]['type']] += data[i]['result']
        recognition_score = OCRBench_score['Regular Text Recognition']+OCRBench_score['Irregular Text Recognition']+OCRBench_score['Artistic Text Recognition']+OCRBench_score['Handwriting Recognition']+OCRBench_score['Digit String Recognition']+OCRBench_score['Non-Semantic Text Recognition']
        Final_score = recognition_score+OCRBench_score['Scene Text-centric VQA']+OCRBench_score['Doc-oriented VQA']+OCRBench_score['Key Information Extraction']+OCRBench_score['Handwritten Mathematical Expression Recognition']
        print("###########################OCRBench##############################")
        print(f"Text Recognition(Total 300):{recognition_score}")
        print("------------------Details of Recognition Score-------------------")
        print(f"Regular Text Recognition(Total 50): {OCRBench_score['Regular Text Recognition']}")
        print(f"Irregular Text Recognition(Total 50): {OCRBench_score['Irregular Text Recognition']}")
        print(f"Artistic Text Recognition(Total 50): {OCRBench_score['Artistic Text Recognition']}")
        print(f"Handwriting Recognition(Total 50): {OCRBench_score['Handwriting Recognition']}")
        print(f"Digit String Recognition(Total 50): {OCRBench_score['Digit String Recognition']}")
        print(f"Non-Semantic Text Recognition(Total 50): {OCRBench_score['Non-Semantic Text Recognition']}")
        print("----------------------------------------------------------------")
        print(f"Scene Text-centric VQA(Total 200): {OCRBench_score['Scene Text-centric VQA']}")
        print("----------------------------------------------------------------")
        print(f"Doc-oriented VQA(Total 200): {OCRBench_score['Doc-oriented VQA']}")
        print("----------------------------------------------------------------")
        print(f"Key Information Extraction(Total 200): {OCRBench_score['Key Information Extraction']}")
        print("----------------------------------------------------------------")
        print(f"Handwritten Mathematical Expression Recognition(Total 100): {OCRBench_score['Handwritten Mathematical Expression Recognition']}")
        print("----------------------Final Score-------------------------------")
        print(f"Final Score(Total 1000): {Final_score}")
    else:
        for i in range(len(data)):
            num_all[data[i]['dataset_name']] += 1
            if data[i].get("result",100)==100:
                continue
            AllDataset_score[data[i]['dataset_name']] += data[i]['result']
        for key in AllDataset_score.keys():
            print(f"{key}: {AllDataset_score[key]/float(num_all[key])}")`
